### PR TITLE
Add stack trace to server lua engine

### LINKED ISSUE #297: Better Lua error handling
Stacktraces would be nice for lua, so that errors are less bad.

### PR DESCRIPTION
**Pull request by SaltySnail**
Don't worry about it

### CODE REVIEW

File: src/TLuaEngine.cpp [hunk 1]
```diff
@@ -30,12 +30,16 @@
...
 #include <nlohmann/json.hpp>
+#include <optional>
 #include <random>
+#include <sol/stack_core.hpp>
 #include <thread>
 #include <tuple>
 
 TLuaEngine* LuaAPI::MP::Engine;
 
+static sol::protected_function addTraceback(sol::state_view StateView, sol::protected_function RawFn);
```
Comment (lionkor) [inline src/TLuaEngine.cpp]:
```suggestion
static sol::protected_function AddTraceback(sol::state_view StateView, sol::protected_function RawFn);
```
Let's keep with the absolutely \*\*\*\*\*\*\*\* naming scheme pls

**Review by lionkor — CHANGES_REQUESTED** [review]
lgtm just one little thing

**Comment by lionkor** [comment]
@SaltySnail you'll have to change it in other places too ;)

File: src/TLuaEngine.cpp [hunk 2]
```diff
@@ -491,6 +495,7 @@ sol::table TLuaEngine::StateThreadData::Lua_TriggerGlobalEvent(const std::string
     sol::variadic_results LocalArgs = JsonStringToArray(Str);
     for (const auto& Handler : MyHandlers) {
         auto Fn = mStateView[Handler];
+        Fn = addTraceback(mStateView, Fn);
```
Comment (SaltySnail) [inline src/TLuaEngine.cpp]:
```suggestion
        Fn = AddTraceback(mStateView, Fn);
```

File: src/TLuaEngine.cpp [hunk 3]
```diff
@@ -1049,6 +1056,21 @@ void TLuaEngine::StateThreadData::RegisterEvent(const std::string& EventName, co
     mEngine->RegisterEvent(EventName, mStateId, FunctionName);
 }
 
+static sol::protected_function addTraceback(sol::state_view StateView, sol::protected_function RawFn) {
```
Comment (SaltySnail) [inline src/TLuaEngine.cpp]:
```suggestion
static sol::protected_function AddTraceback(sol::state_view StateView, sol::protected_function RawFn) {
```

File: src/TLuaEngine.cpp [hunk 4]
```diff
@@ -1149,6 +1171,7 @@ void TLuaEngine::StateThreadData::operator()() {
                             break;
                         }
                     }
+                    auto Fn = addTraceback(StateView, RawFn);
```
Comment (SaltySnail) [inline src/TLuaEngine.cpp]:
```suggestion
                    auto Fn = AddTraceback(StateView, RawFn);
```

**Review by SaltySnail — COMMENTED** [review]
Fixed not following naming convention in other places. Sorry for the atomic commits, was on my phone

**Comment by lionkor** [comment]
WTF lmao

**Review by lionkor — APPROVED** [review]

**Pull request merged**